### PR TITLE
Codereview session 2. 

### PR DESCRIPTION
**Pull request by ramrunner**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+### Import Style
+All imports within the project should be contained within an import() block, even if there's only one. Gofmt will sort imports
+within a group alphabetically. Import groups are separated by a newline. There should only be 3 import groups, appearing in order.
+
+1. Standard library
+2. Local project packages
+3. 3rd party libraries
+
+Some files may not use all of the groups above. Here is an example of the import style:
+
```
Comment (natemort) [inline CONTRIBUTING.md]: If you use "\```go" at the start and "\```" at the end of your code snippets then you can get syntax highlighting.

```go
    import (
        "fmt"
        "os"

        "github.com/CSUNetSec/bgpmon/util"

        "github.com/spf13/cobra"
    )
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+In golang, it is possible to declare mulitple variable in the conditional line of an if statement. The scope of this variable 
+is the extent of the if and any attached else clauses.
+
+    if a, b := foo; a != b {
+        // a and b are in scope here
+    } else {
+        // Also in scope here
+    }
+    // Neither are in scope here
+This type of statement is **not** allowed in BGPMon. Instead, the following style is preferred:
```
Comment (natemort) [inline CONTRIBUTING.md]: I was under the impression that we settled on this standard specifically just for error handling rather than a general thing, as described [here](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow).  At the same time I don't think this situation happens for any non-error types anywhere in the project, so it doesn't really matter that much.

File: cmd/bgpmon/cmd/root.go [hunk 3]
```diff
@@ -79,7 +79,7 @@ func Execute() {
 func init() {
 	cobra.OnInitialize(initConfig)
 	rootCmd.PersistentFlags().StringVarP(&cfgFile, "config", "c", "$HOME/.bgpmon.yaml", "Config file")
-	rootCmd.PersistentFlags().StringVarP(&bgpmondHost, "host", "h", "127.0.0.1", "bgpmond host to connect to")
+	rootCmd.PersistentFlags().StringVarP(&bgpmondHost, "host", "H", "127.0.0.1", "bgpmond host to connect to")
```
Comment (natemort) [inline cmd/bgpmon/cmd/root.go]: What's the reason for this change?

File: cmd/bgpmon/cmd/write.go [hunk 4]
```diff
@@ -5,34 +5,37 @@ import (
...
 	"github.com/spf13/cobra"
 )
 
 var writeCmd = &cobra.Command{
-	Use:   "write SESS_ID [-f filtFile] files...",
-	Short: "writes a series of files to a session ID",
-	Long:  "Given a series of files, write will write up to <worker count> of them at a single time, reporting the individual and total success upon completion.",
+	Use:   "write SESS_ID FILES...",
+	Short: "Writes BGP captures from a file(s) to a session.",
+	Long:  "Opens a write stream(s) on the provided session and writes <workers> files concurrently. Write generates a report upon completion of the success or failure of individual files",
```
Comment (natemort) [inline cmd/bgpmon/cmd/write.go]: Missing period.

File: cmd/bgpmon/cmd/write.go [hunk 5]
```diff
@@ -68,109 +75,105 @@ func writeFunc(cmd *cobra.Command, args []string) {
...
-			}
-
-			tot++
-			if r.err != nil {
-				failCt++
-				failed = append(failed, r.fname)
-				reasons = append(reasons, r.err)
-			}
+	numWritten := 0
+	numFailed := 0
```
Comment (natemort) [inline cmd/bgpmon/cmd/write.go]: numFailed could be replaced with len(failed).

File: cmd/bgpmon/cmd/root.go (hunk 3 above)
Comment (WillYingling) [reply cmd/bgpmon/cmd/root.go]: The cobra library panics when you try to override the "h" flag, because it considers it pre-defined as the help flag. To maintain consistency, we were going to change all of the global flags to be capitalized.

File: CONTRIBUTING.md (hunk 2 above)
Comment (WillYingling) [reply CONTRIBUTING.md]: We can discuss this more in the review session this week, but I'd prefer to make our standards as general as possible, rather than defining them for each use case. As you said, this doesn't really create an issue for the rest of the project, so I'll leave this as-is for now.

File: cmd/bgpmon/cmd/root.go (hunk 3 above)
Comment (natemort) [reply cmd/bgpmon/cmd/root.go]: Sounds good to me.

**Comment by natemort** [comment]
Looks good to me.
